src/Microsoft.MobileBlazorBindings.Hosting/WebViewIPC.cs

```diff
@@ -0,0 +1,243 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Components.RenderTree;
+using Microsoft.JSInterop;
+using Microsoft.JSInterop.Infrastructure;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.MobileBlazorBindings.Hosting
+{
+    public class WebViewIPC : IDisposable
+    {
+        private readonly IWebViewIPCAdapter _webViewIPCAdapter;
+        private readonly Dispatcher _dispatcher;
+
+        private readonly Dictionary<string, List<Action<object>>> _registrations = new();
```

Comment:
Would it make sense to use ConcurrentDictionary instead?

Reply:
This is all setup code for the IPC system, so it's all called once from `AttachInteropAsync` in a single thread.

Reply:
Why do you need locks then?

Reply:
Well that's a good question, and I don't know the answer. But we will rewrite this entire class anyway when it moves into the production repo. We'll have some more Blazor experts take a look at this and make sure we have the right patterns. I'm guessing this would be fine in its _current_ usage without any locking because it's all called from 1 thread at 1 point in time. But maybe this code (not written by me) was originally intended to handle other cases that aren't needed here.